src/Microsoft.SqlTools.ServiceLayer/Scripting/Contracts/ScriptingOptions.cs

```diff
@@ -0,0 +1,178 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.SqlTools.ServiceLayer.Scripting.Contracts
+{
+    /// <summary>
+    /// Database scripting options
+    /// </summary>
+    public class ScriptOptions
+    {
+        /// <summary>
+        /// Generate ANSI padding statements
+        /// </summary>
+        public string ScriptAnsiPadding { get; set; }
```

Comment:
These can be null if they aren't specified in the request. I can explore making them Nullable<bool>, I just wasn't sure if the JSON-RPC layer handles this, but I'll look into it.

---
In reply to: [111652013](https://github.com/Microsoft/sqltoolsservice/pull/316#discussion_r111652013) [](ancestors = 111652013)